xdsl/utils/scoped_dict.py

```diff
@@ -0,0 +1,62 @@
+from __future__ import annotations
+
+from typing import Generic, TypeVar
+
+_Key = TypeVar("_Key")
```

Comment:
Should this be hashable or something? I'm not sure why pyright doesn't complain here

Reply:
Hashable isn't a real type, AFAIK, and the `dict` generic type has the same `TypeVar` definition